src/input/mod.rs

```diff
@@ -129,32 +127,42 @@ impl<'a> InputFile<'a> {
     }
 }
 
-impl<'a> Drop for InputFile<'a> {
-    fn drop(&mut self) {
-        unsafe { CEXR_InputFile_delete(self.handle) };
-    }
-}
-
-struct IStream<'a> {
-    handle: *mut CEXR_IStream,
-    _phantom: PhantomData<&'a ()>,
-}
+impl<'a> InputFile<'a, &'a [u8]> {
```

Comment:
If we don't, then you have to specify the type to access the method, like so:

```
let f = InputFile::<&[u8]>::from_slice(/* params */);
```

If you just do this:

```
let f = InputFile::from_slice(/* params */);
```

It won't compile because the compiler can't infer `T`, because for `from_slice` T is an output type parameter not an input type parameter.

However, I'm wondering if we even really need T as part of InputFile and OutputFile now.  If remove those, then it shouldn't be a problem.  We can make the phantom type be a reference to `()` or `[u8]` or something.